hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/TestStorageContainerManager.java

```diff
@@ -289,153 +339,74 @@ public void testBlockDeletionTransactions() throws Exception {
-    conf.setTimeDuration(HDDS_HEARTBEAT_INTERVAL, 1000, TimeUnit.MILLISECONDS);
-    conf.setTimeDuration(ScmConfigKeys.OZONE_SCM_HEARTBEAT_PROCESS_INTERVAL, 3000, TimeUnit.MILLISECONDS);
-
-
-
-    try (MiniOzoneCluster cluster = MiniOzoneCluster.newBuilder(conf)
-        .setNumDatanodes(1)
-        .build()) {
-      cluster.waitForClusterToBeReady();
-      HddsDatanodeService datanode = cluster.getHddsDatanodes().get(0);
-      StorageContainerManager scm = cluster.getStorageContainerManager();
-      File dbDir = scm.getScmMetadataStore().getStore().getDbLocation();
-      scm.stop();
-
-      // re-initialise SCM with new clusterID
-
-      GenericTestUtils.deleteDirectory(new File(SCMHAUtils.getRatisStorageDir(conf)));
-      GenericTestUtils.deleteDirectory(dbDir);
-      GenericTestUtils.deleteDirectory(
-          new File(scm.getScmStorageConfig().getStorageDir()));
-      String newClusterId = UUID.randomUUID().toString();
-      StorageContainerManager.scmInit(scm.getConfiguration(), newClusterId);
-      scm = HddsTestUtils.getScmSimple(scm.getConfiguration());
-
-      DatanodeStateMachine dsm = datanode.getDatanodeStateMachine();
+  // assumes SCM is already stopped
+  private void testOldDNRegistersToReInitialisedSCM(MiniOzoneCluster cluster) throws Exception {
+    HddsDatanodeService datanode = cluster.getHddsDatanodes().get(0);
+
+    // re-initialise SCM with new clusterID
+    String newClusterId = UUID.randomUUID().toString();
+    StorageContainerManager.scmInit(cluster.getConf(), newClusterId);
+    StorageContainerManager scm = HddsTestUtils.getScmSimple(cluster.getConf());
+
+    DatanodeStateMachine dsm = datanode.getDatanodeStateMachine();
+    assertEquals(DatanodeStateMachine.DatanodeStates.RUNNING,
+        dsm.getContext().getState());
+    // DN Endpoint State has already gone through GetVersion and Register,
+    // so it will be in HEARTBEAT state.
+    for (EndpointStateMachine endpoint : dsm.getConnectionManager()
+        .getValues()) {
+      assertEquals(EndpointStateMachine.EndPointStates.HEARTBEAT,
+          endpoint.getState());
+    }
+    GenericTestUtils.LogCapturer scmDnHBDispatcherLog =
+        GenericTestUtils.LogCapturer.captureLogs(
+            SCMDatanodeHeartbeatDispatcher.LOG);
+    LogManager.getLogger(HeartbeatEndpointTask.class).setLevel(Level.DEBUG);
+    GenericTestUtils.LogCapturer heartbeatEndpointTaskLog =
+        GenericTestUtils.LogCapturer.captureLogs(HeartbeatEndpointTask.LOG);
+    GenericTestUtils.LogCapturer versionEndPointTaskLog =
+        GenericTestUtils.LogCapturer.captureLogs(VersionEndpointTask.LOG);
+    // Initially empty
+    assertThat(scmDnHBDispatcherLog.getOutput()).isEmpty();
+    assertThat(versionEndPointTaskLog.getOutput()).isEmpty();
+    // start the new SCM
+    try {
+      scm.start();
+      // Initially DatanodeStateMachine will be in Running state
       assertEquals(DatanodeStateMachine.DatanodeStates.RUNNING,
```

Comment:
This should be removed. It is moved to Line 355.

Reply:
Good point, thanks.